src/wallet/tx_builder.rs

```diff
@@ -32,54 +32,118 @@
 pub trait TxBuilderContext: std::fmt::Debug + Default + Clone {}
 
-/// [`Wallet::create_tx`](super::Wallet::create_tx) context
+/// Marker type to indicate the [`TxBuilder`] is being used to create a new transaction (as opposed
+/// to bumping the fee of an existing one).
 #[derive(Debug, Default, Clone)]
 pub struct CreateTx;
 impl TxBuilderContext for CreateTx {}
 
-/// [`Wallet::bump_fee`](super::Wallet::bump_fee) context
+/// Marker type to indicate the [`TxBuilder`] is being used to bump the fee of an existing transaction.
 #[derive(Debug, Default, Clone)]
 pub struct BumpFee;
 impl TxBuilderContext for BumpFee {}
 
 /// A transaction builder
 ///
-/// This structure contains the configuration that the wallet must follow to build a transaction.
+/// A `TxBuilder` is initially created by calling [`build_tx`] or [`build_fee_bump`] on a wallet.
+/// From there you set sepcific options on the builder until finally calling [`finish`] to get the transaction.
 ///
-/// For an example see [this module](super::tx_builder)'s documentation;
-#[derive(Debug)]
-pub struct TxBuilder<D: Database, Cs: CoinSelectionAlgorithm<D>, Ctx: TxBuilderContext> {
+/// Each method on TxBuilder takes and returns `&mut self` so you can use either use a chaining call
+/// or assign the builder and call normally as in the following example:
+///
+/// ```
+/// # use bdk::*;
+/// # use bdk::wallet::tx_builder::*;
+/// # use bitcoin::*;
+/// # use core::str::FromStr;
+/// # let wallet = doctest_wallet!();
+/// # let addr1 = Address::from_str("2N4eQYCbKUHCCTUjBJeHcJp9ok6J2GZsTDt").unwrap();
+/// # let addr2 = addr1.clone();
+/// // chaining
+/// let (psbt1, details) =  wallet.build_tx()
+///        .ordering(TxOrdering::Untouched)
+///        .add_recipient(addr1.script_pubkey(), 50_000)
+///        .add_recipient(addr2.script_pubkey(), 50_000)
+///        .finish()?;
+///
+/// // non-chaining
+/// let mut builder = wallet.build_tx();
+/// for addr in &[addr1, addr2] {
+///     builder.add_recipient(addr.script_pubkey(), 50_000);
+/// }
+/// let (psbt2, details) = builder.ordering(TxOrdering::Untouched).finish()?;
+/// //
+/// assert_eq!(psbt1.global.unsigned_tx.output[..2], psbt2.global.unsigned_tx.output[..2]);
+/// # Ok::<(), bdk::Error>(())
+/// ```
+///
+/// At the moment [`coin_selection`] is an exception - it consumes `self`.
+/// This means it is usually best to call [`coin_selection`] first before calling other methods.
+///
+/// Note that calling methods on the builder after calling [`finish`] will result in a panic.
+///
+/// For further examples see [this module](super::tx_builder)'s documentation;
+///
+/// [`build_tx`]: Wallet::build_tx
+/// [`build_fee_bump`]: Wallet::build_fee_bump
+/// [`finish`]: Self::finish
+/// [`coin_selection`]: Self::coin_selection
+pub struct TxBuilder<'a, B, D, Cs, Ctx> {
+    pub(crate) wallet: &'a Wallet<B, D>,
+    // params and coin_selection are Options not becasue they are optionally set (they are always
```

Comment:
Helpful explainer, but one little typo, becasue => because

Reply:
option stuff is gone.